test/pool.test.ts

```diff
@@ -116,6 +148,19 @@ async function withdraw(
   await expectWithdrawable(pool, amountTo);
 }
 
+async function expectSetReceiverReverts(
+  pool: Pool,
+  address: string,
+  weight: number
+): Promise<void> {
+  try {
+    await pool.setReceiver(address, weight);
+  } catch {
```

Comment:
We should probably inspect the error to ensure that we’re only catching the correct ones and re-throw others.

Reply:
Fixed